core/src/main/java/org/mule/runtime/core/internal/context/FlowStoppedPersistenceListener.java

```diff
@@ -36,4 +36,9 @@ public interface FlowStoppedPersistenceListener {
    * Indicates if the flow should start or not
    */
   Boolean shouldStart();
+
+  /**
+   * Indicates if the initial state of the flow should be ignored.
+   */
+  Boolean ignoreInitialState();
```

Comment:
I don't think this should be add as a property to persist. The persisted state of the flows should reflect the current status of the flows. Then according to the feature flag you have to decide if that is taken into account or not.